src/asound/bluealsa-pcm.c

```diff
@@ -345,10 +348,19 @@ static void *io_thread(snd_pcm_ioplug_t *io) {
 
 fail:
 
+	if (pcm->ba_pcm_fd != -1) {
+		close(pcm->ba_pcm_fd);
+		pcm->ba_pcm_fd = -1;
+	}
+	if (pcm->ba_pcm_ctrl_fd != -1) {
+		close(pcm->ba_pcm_ctrl_fd);
+		pcm->ba_pcm_ctrl_fd = -1;
+	}
+
+	/* make sure we will not get stuck in the pause sync loop */
 	pthread_mutex_lock(&pcm->mutex);
-	pcm->pause_state = BA_PAUSE_STATE_PAUSED;
+	pcm->connected = false;
```

Comment:
I would keep the `pcm->pause_state = BA_PAUSE_STATE_PAUSED` like it was before. I think that the `pause_state` is more related to the `pause_cond` than the `connected` fag. But since there is now a `connected` flag I would set it to false just after "disconnected" notification from `read()`/`write()` - just before `goto fail` in lines 307, 314 and 328. In that way it will be easier to read/understand the flow IMHO.